api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java

```diff
@@ -51,16 +71,19 @@ public boolean isInitialized() {
 	 * @should throw an error if generated identifier is longer than maxLength
      */
     public String getIdentifierForSeed(long seed) {
-
     	// Convert the next sequence integer into a String with the appropriate Base characters
 		int seqLength = firstIdentifierBase == null ? 1 : firstIdentifierBase.length();
 
 		String identifier = IdgenUtil.convertToBase(seed, baseCharacterSet.toCharArray(), seqLength);
-
-    	// Add optional prefix and suffix
-    	identifier = (prefix == null ? identifier : prefix + identifier);
+		// Check whether we have a PrefixProvider
+		PrefixProvider prefixProvider = getPrefixProvider(prefix);
+		if (prefixProvider != null) {
```

Comment:
When is `prefixProvider` ever null, can this happen?

Reply:
Yes it's possible. that is if the `getPrefixProvider(String)` method is given a null argument. This is a case where we have a null `prefix`.